src/lib.rs

```diff
@@ -2777,6 +2777,161 @@ pub fn is_maximal_matching(
+        );
+
+    match triangles {
+        0 => 0.0,
+        _ => triangles as f64 / triples as f64,
+    }
+}
+
+fn _digraph_triangles(
+    graph: &digraph::PyDiGraph,
+    node: usize,
+) -> (usize, usize) {
+    let mut triangles: usize = 0;
+
+    let index = NodeIndex::new(node);
+    let mut out_neighbors: HashSet<NodeIndex> = graph
+        .graph
+        .neighbors_directed(index, petgraph::Direction::Outgoing)
+        .collect();
+    out_neighbors.remove(&index);
+
+    let mut in_neighbors: HashSet<NodeIndex> = graph
+        .graph
+        .neighbors_directed(index, petgraph::Direction::Incoming)
+        .collect();
+    in_neighbors.remove(&index);
+
+    let neighbors = out_neighbors.iter().chain(in_neighbors.iter());
+
+    for nodev in neighbors {
+        triangles += graph
+            .graph
+            .neighbors_directed(*nodev, petgraph::Direction::Outgoing)
+            .chain(
+                graph
+                    .graph
+                    .neighbors_directed(*nodev, petgraph::Direction::Incoming),
+            )
+            .map(|x| {
+                let mut res: usize = 0;
+
+                if (x != *nodev) && out_neighbors.contains(&x) {
+                    res += 1;
+                }
+                if (x != *nodev) && in_neighbors.contains(&x) {
+                    res += 1;
+                }
+                res
+            })
+            .sum::<usize>();
+    }
+
+    let din: usize = in_neighbors.len();
+    let dout: usize = out_neighbors.len();
+
+    let dtot = dout + din;
+    let dbil: usize = out_neighbors.intersection(&in_neighbors).count();
+    let triples: usize = match dtot {
+        0 => 0,
+        _ => dtot * (dtot - 1) - 2 * dbil,
+    };
+
+    (triangles / 2, triples)
+}
+
+/// Compute the transitivity of a directed graph.
+///
+/// The transitivity of a directed graph is defined in [Fag]_, Eq.8:
+///
+/// .. math::
+///     `c=3 \times \frac{\text{number of triangles}}{\text{number of all possible triangles}}`
+///
+/// A triangle is a connected triple of nodes.
+/// Different edge orientations counts as different triangles.
+///
+/// It's worh noting that the function implicitly assumes that there is
+/// no parallel edges or self loops.
+///
+/// :param PyDiGraph graph: Directed graph to be used.
+///
+/// :returns: Transitivity.
+/// :rtype: float
+///
+/// .. [Fag] Clustering in complex directed networks by G. Fagiolo,
+///    Physical Review E, 76(2), 026107 (2007)
+#[pyfunction]
+#[text_signature = "(graph, /)"]
+fn digraph_transitivity(graph: &digraph::PyDiGraph) -> f64 {
+    let node_indices: Vec<NodeIndex> = graph.graph.node_indices().collect();
+    let (triangles, triples) = node_indices
+        .par_iter()
+        .map(|node| _digraph_triangles(graph, node.index()))
+        .reduce(
+            || (0, 0),
+            |(sumx, sumy), (resx, resy)| (sumx + resx, sumy + resy),
+        );
+
+    match triangles {
+        0 => 0.0,
```

Comment:
Coverage reporting showed that there wasn't any coverage for this line: https://coveralls.io/builds/38209715/source?filename=src%2Flib.rs#L2930

Not critical because it's hard to imagine this line misbehaving, but it might be good to add an empty graph test case for digraph too so we execute this line too.